src/super_gradients/training/models/segmentation_models/segformer.py

```diff
@@ -0,0 +1,516 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from super_gradients.training.models import HpmStruct
+from super_gradients.training.utils import get_param
+from super_gradients.training.models.segmentation_models.segmentation_module import SegmentationModule
+from super_gradients.training.utils.regularization_utils import DropPath
+
+from typing import List, Tuple
+
+"""
+paper:  SegFormer: Simple and Efficient Design for Semantic Segmentation with Transformers
+        ( https://arxiv.org/pdf/2105.15203.pdf )
+code adopted from git repo: https://github.com/sithu31296/semantic-segmentation
+
+Imagenet-1k pre-trained backbone weights taken and adapted from: https://github.com/sithu31296/semantic-segmentation
+
+"""
+
+
+class PatchEmbedding(nn.Module):
+    def __init__(self, in_channels: int, out_channels: int, patch_size: int, stride: int, padding: int):
+        """
+        Overlapped patch merging (https://arxiv.org/pdf/2105.15203.pdf)
+        :param in_channels: number of input channels
+        :param out_channels: number of output channels (embedding dimension)
+        :param patch_size: patch size (k for size (k, k))
+        :param stride: patch stride (k for size (k, k))
+        :param padding:  patch padding (k for size (k, k))
+        """
+
+        super().__init__()
+
+        self.proj = nn.Conv2d(in_channels=in_channels, out_channels=out_channels, kernel_size=patch_size, stride=stride, padding=padding)
+        self.norm = nn.LayerNorm(out_channels)
+
+    def forward(self, x: torch.Tensor) -> Tuple[torch.Tensor, int, int]:
+        x = self.proj(x)
+        _, _, h, w = x.shape
+
+        x = x.flatten(2).transpose(1, 2)
+        x = self.norm(x)
+
+        return x, h, w
+
+
+class EfficientSelfAttention(nn.Module):
+    def __init__(self, dim: int, head: int, sr_ratio: int):
+        """
+        Efficient self-attention (https://arxiv.org/pdf/2105.15203.pdf)
+        :param dim: embedding dimension
+        :param head: number of attention heads
+        :param sr_ratio: the reduction ratio of the efficient self-attention
+        """
+
+        super().__init__()
+
+        self.head = head
+        self.sr_ratio = sr_ratio
+        self.scale = (dim // head) ** -0.5
+        self.q = nn.Linear(dim, dim)
+        self.kv = nn.Linear(dim, dim * 2)
+        self.proj = nn.Linear(dim, dim)
+
+        if sr_ratio > 1:
+            self.sr = nn.Conv2d(dim, dim, sr_ratio, sr_ratio)
+            self.norm = nn.LayerNorm(dim)
+
+    def forward(self, x: torch.Tensor, h: int, w: int) -> torch.Tensor:
+        b, n, c = x.shape
+        q = self.q(x).reshape(b, n, self.head, c // self.head).permute(0, 2, 1, 3)
```

Comment:
Did this code successfully convert to ONNX?

Reply:
The whole model was succesfully converted to ONNX, but I didn't check the performance of the ONNX model